packages/react-form/README.md

```diff
@@ -0,0 +1,873 @@
+                </Card.Section>
+              </Card>
+            </Layout.Section>
+          </Layout>
+        </Page>
+      </Form>
+    </Frame>
+  );
+}
+```
+
+You can also configure fields ahead of time and pass them in to `useForm` afterwards. This is useful when you need one field to depend upon another for validation.
+
+```tsx
+import * as React from 'react';
+import {
+  Page,
+  TextField,
+  FormLayout,
+  Card,
+  Layout,
+  Form,
+  ContextualSaveBar,
+  Frame,
+  Banner,
+} from '@shopify/polaris';
+import {useField, useForm} from '@shopify/react-form';
+
+function MyComponent() {
+  const title = useField('');
+  const price = useField(
+    {
+      value: '0.00',
+      validates: value => {
+        if (title.value.includes('expensive') && parseFloat(value) < 1000) {
+          return 'Expensive items must cost more than 1000 dollars';
+        }
+      },
+    },
+    [title.value],
+  );
+
+  const {submit, submitting, dirty, reset, submitErrors} = useForm({
+    fields: {title, description},
+    onSubmit: fieldValues => {
+      return {status: 'fail', errors: [{message: 'bad form data'}]};
+    },
+  });
+
+  const contextBar = dirty ? (
+    <ContextualSaveBar
+      message="New product"
+      saveAction={{
+        onAction: submit,
+        loading: submitting,
+        disabled: false,
+      }}
+      discardAction={{
+        onAction: reset,
+      }}
+    />
+  ) : null;
+
+  const errorBanner =
+    submitErrors.length > 0 ? (
+      <Layout.Section>
+        <Banner status="critical">
+          <p>There were some issues with your form submission:</p>
+          <ul>
+            {submitErrors.map(({message}, index) => {
+              return <li key={`${message}${index}`}>{message}</li>;
+            })}
+          </ul>
+        </Banner>
+      </Layout.Section>
+    ) : null;
+
+  return (
+    <Frame>
+      <Form onSubmit={submit}>
+        <Page title={pageTitle}>
+          {contextBar}
+          <Layout>
+            {errorBanner}
+            <Layout.Section>
+              <Card>
+                <Card.Section>
+                  <FormLayout>
+                    <TextField label="Title" {...title} />
+                    <TextField type="number" label="Price" {...price} />
+                  </FormLayout>
+                </Card.Section>
+              </Card>
+            </Layout.Section>
+          </Layout>
+        </Page>
+      </Form>
+    </Frame>
+  );
+}
+```
+
+##### Remarks
+
+- **Building your own:** Internally, `useForm` is a convenience wrapper over `useDirty`, `useReset`, `useValidateAll`, `useSubmit`, and `useErrorPropagation`.If you only need some of its functionality, consider building a custom hook combining a subset of them.
+- **Subforms:** You can have multiple `useForm`s wrapping different subsets of a group of fields. Using this you can submit subsections of the form independently and have all the error and dirty tracking logic "just work" together.
+
+####useSubmit
```

Comment:
You need a line break between these otherwise they end up on one unformatted line.